backend/examples/verifying_inclusion.rs

```diff
@@ -0,0 +1,69 @@
+use std::fs::File;
+use std::io::prelude::*;
+
+use halo2_proofs::halo2curves::{bn256::Fr as Fp, ff::PrimeField};
+use num_bigint::BigInt;
+
+use summa_backend::apis::utils::generate_setup_artifacts;
+use summa_solvency::{
+    circuits::{merkle_sum_tree::MstInclusionCircuit, utils::full_verifier},
+    merkle_sum_tree::Entry,
+};
+
+fn generate_leaf_hash<const N_ASSETS: usize>(user_name: String, balances: Vec<usize>) -> Fp {
+    // Convert usize to BigInt for the `Entry` struct
+    let balances_big_int: Vec<BigInt> = balances.into_iter().map(BigInt::from).collect();
+
+    let entry: Entry<N_ASSETS> =
+        Entry::new(user_name, balances_big_int.try_into().unwrap()).unwrap();
+
+    entry.compute_leaf().hash
+}
+
+fn main() {
+    const LEVELS: usize = 4;
+    const L: usize = 6;
+    const N_ASSETS: usize = 2;
+
+    // When verifying inclusion proof on user side,
+    // have to load two files, ptau and proof.
+    let ptau_path = "./ptau/hermez-raw-11";
+
+    let mut file = File::open("examples/entry_0_proof.bin").unwrap();
```

Comment:
What if rather than using a the proof in binary file, we build the whole Snapshot here in the example and generate the proof of inclusion for that user using the method on Snapshot? In this way this can serve as an example on how to use the whole Snapshot data structure

For the verification we also fetch the verification key from the Snapshot

Reply:
Updated in [d43498c](https://github.com/summa-dev/summa-solvency/pull/118/commits/d43498c5930568526a2a67a71f9dd146b325edd0)

> For the verification we also fetch the verification key from the Snapshot

However, I cannot envision the user getting the Snapshot instance. 
so, I've added a comment to explain this on [79f37a2](https://github.com/summa-dev/summa-solvency/pull/118/commits/79f37a203b36874dc4cd9138295eebd75dbce92d)